File: app/src/main/java/com/example/solar_alarm/Data/Daos/StaticDataDao.java
package com.example.solar_alarm.Data.Daos;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.solar_alarm.Data.Tables.OffsetType;
import com.example.solar_alarm.Data.Tables.SolarTimeType;

@Dao
public interface StaticDataDao
{
    @Insert
    void Insert(OffsetType offsetType);

    @Query("SELECT EXISTS(SELECT * FROM OffsetTypes)")
    boolean isOffsetTypesExists();

    @Insert
    void Insert(SolarTimeType solarTimeType);

    @Query("SELECT EXISTS(SELECT * FROM SolarTimeTypes)")
    boolean isSolarTimeTypesExists();
}

File: app/src/main/java/com/example/solar_alarm/Data/Daos/StaticDataDao.kt
package com.example.solar_alarm.data.daos

import com.example.solar_alarm.data.enums.OffsetTypeEnum
import com.example.solar_alarm.data.enums.SolarTimeTypeEnum
import androidx.room.*

@Dao
interface StaticDataDao {
    @Insert
    suspend fun insert(offsetType: OffsetTypeEnum)

    @Query("SELECT EXISTS(SELECT * FROM OffsetTypeEnum)")
    suspend fun isOffsetTypesExists(): Boolean

    @Insert
    suspend fun insert(solarTimeType: SolarTimeTypeEnum)

    @Query("SELECT EXISTS(SELECT * FROM SolarTimeTypeEnum)")
    suspend fun isSolarTimeTypesExists(): Boolean
}